ome_zarr/writer.py

```diff
@@ -107,26 +107,38 @@ def _validate_plate_acquisitions(
     return acquisitions
 
 
+def _validate_plate_rows_columns(
+    rows_or_columns: List[str],
+    fmt: Format = CurrentFormat(),
+) -> List[dict]:
+
+    if len(set(rows_or_columns)) != len(rows_or_columns):
+        raise ValueError(f"{rows_or_columns} must contain unique elements")
+    validated_list = []
+    for element in rows_or_columns:
+        if not element.isalnum():
+            raise ValueError(f"{element} must contain alphanumeric characters")
+        validated_list.append({"name": str(element)})
+    return validated_list
+
+
 def _validate_plate_wells(
-    wells: List[Union[str, dict]], fmt: Format = CurrentFormat()
+    wells: List[Union[str, dict]],
+    rows: List[str],
+    columns: List[str],
+    fmt: Format = CurrentFormat(),
 ) -> List[dict]:
 
-    VALID_KEYS = [
-        "path",
-    ]
     validated_wells = []
     if wells is None or len(wells) == 0:
         raise ValueError("Empty wells list")
     for well in wells:
         if isinstance(well, str):
-            validated_wells.append({"path": str(well)})
+            well_dict = fmt.generate_well_dict(well, rows, columns)
+            fmt.validate_well_dict(well_dict)
+            validated_wells.append(well_dict)
         elif isinstance(well, dict):
-            if any(e not in VALID_KEYS for e in well.keys()):
-                LOGGER.debug("f{well} contains unspecified keys")
-            if "path" not in well:
-                raise ValueError(f"{well} must contain a path key")
-            if not isinstance(well["path"], str):
-                raise ValueError(f"{well} path must be of str type")
+            fmt.validate_well_dict(well)
```

Comment:
I wonder if you want to validate that `well` against the rows and columns?
E.g. this is invalid, but I think wouldn't fail:
```
_validate_plate_wells([{"path": "Z/10"}], ["A"], ["1"])
```

Reply:
This is probably a point up for discussion. As  per the current specification proposal, what you suggest is _not_ invalid i.e. there is no enforced relationship between the Zarr group names and the rows/columns names. Instead the relationship is purely expressed via the row and column indices.

In the typical usage, I also expect these groups and names should match and this is what the row/column index computation logic assumes.

I'll push another unit test to cover this use case shortly

Reply:
See 16986d0

Reply:
The spec PR says "The `path` MUST consist of a `name` in the `rows` list, a file separator (`/`),
      and a `name` from the `columns` list, in that order."

Reply:
Yes, I just read that and I commented on the spec PR itself - https://github.com/ome/ngff/pull/24#discussion_r597699471.